### ISSUE TITLE
False positive `used-before-assignment` when TYPE_CHECKING is used with if/elif/else blocks

### ISSUE THREAD

**Issue by yilei** [PyCQA/pylint#7574]
### Bug description

Example

```python
# pylint: disable=import-error,missing-function-docstring,invalid-name
"""."""

from typing import TYPE_CHECKING, Optional
from package import runtime


if TYPE_CHECKING:
    from package import module
elif runtime.is_local:
    from package import module
else:
    module = None


def diagnose() -> Optional[module.Result]:
    if module is not None:
        return module.diagnose_result()
    return None
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:16:27: E0601: Using variable 'module' before assignment (used-before-assignment)

------------------------------------------------------------------
Your code has been rated at 5.45/10 (previous run: 5.45/10, +0.00)
```

### Expected behavior

No `used-before-assignment` should be raised.

### Pylint version

```shell
pylint 2.15.3
astroid 2.12.10
Python 3.8.10 (default, Jul  6 2022, 10:42:32)
[Clang 13.0.0 (clang-1300.0.29.30)]
```
